crates/wac-parser/src/resolution.rs

```diff
@@ -681,6 +681,44 @@ pub enum Error {
         #[label(primary, "spreading the exports of this instance has no effect")]
         span: SourceSpan,
     },
+    /// An import is not in the target world.
+    #[error("import `{name}` is not present in target world `{world}`")]
+    ImportNotInTarget {
+        /// The import name.
+        name: String,
+        /// The target world.
+        world: String,
+        /// The span where the error occurred.
+        #[label(primary, "import `{name}` is not in the target world")]
+        span: SourceSpan,
+    },
+    /// Missing an export for the target world.
+    #[error("missing export `{name}` for target world `{world}`")]
+    MissingTargetExport {
+        /// The export name.
+        name: String,
+        /// The target world.
+        world: String,
+        /// The span where the error occurred.
+        #[label(primary, "must export `{name}` to target this world")]
```

Comment:
```suggestion
        #[label(primary, "must export a {kind} named `{name}` to target this world")]
```

Where `kind` is set to the expected item kind.